app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -271,8 +271,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
             // if handling new return code (want to return to home screen) but a return at the end of your statement
             switch(requestCode) {
             case PREFERENCES_ACTIVITY:
-                // rebuild buttons in case language was changed
-                uiController.setupUI();
+                if (resultCode == RESULT_CANCELED) {
+                    // rebuild home buttons in case language changed;
+                    // but only if we didn't just clear user data
+                    uiController.setupUI();
```

Comment:
Hm, there are a lot of ways for us to convey this. Can't we just return a different result code for clearing data, like RESULT_DATA_RESET and capture around that?